demos/run_scripts.sh

```diff
@@ -31,14 +31,19 @@ GIT_CONFIG_GLOBAL=$(mktemp --tmpdir gitconfig-XXXX)
 export GIT_CONFIG_GLOBAL
 git_config "$GIT_CONFIG_GLOBAL"
 
-for script in "$@"; do
+run_script_through_term_transcript_and_pipe_result_to_stderr() {
+  script="$1"
   script_base="${script%.sh}"
   script_base="${script_base#demo_}"
   # We use `term-transcript capture` instead of `term-transcript exec` so that
   # we can show the output of the script via `tee`.
   (bash "$script" || (echo "SCRIPT FAILED WITH EXIT CODE $?"; false)) 2>&1 | \
     tee /dev/stderr | \
     term-transcript capture --no-inputs --pure-svg --out "$script_base".svg "$script_base"
+}
```

Comment:
Would it work to store the output in a variable and not pass it back and forth via stdout and stderr? Something like this (with this function inlined again):

```
  output="$(bash "$script")" || (echo "SCRIPT FAILED WITH EXIT CODE $?"; false)
  echo "$output" | term-transcript capture --no-inputs --pure-svg --out "$script_base".svg "$script_base"
  echo "$output"
```

Reply:
As written, this wouldn't print the output when there's an error in the script. If `$output` is available in the `||` block, I could print it there I suppose.

Perhaps it'd be cleaner to use a temporary file.

Also, either of these options would suppress the streaming output (where you see how long different commands take), but I'm not sure that's valuable.

Reply:
It does seem a bit useful to have streaming output. That should be possible by teeing to a temporary file (which is maybe why you mentioned it)